src/parser_impl.c

```diff
@@ -0,0 +1,125 @@
+/*******************************************************************************
+*  (c) 2019 Zondax GmbH
+*
+*  Licensed under the Apache License, Version 2.0 (the "License");
+*  you may not use this file except in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+********************************************************************************/
+
+#include "parser_impl.h"
+
+parser_tx_t parser_tx_obj;
+
+parser_error_t parser_init_context(parser_context_t *ctx,
+                                   const uint8_t *buffer,
+                                   uint16_t bufferSize) {
+    ctx->offset = 0;
+
+    if (bufferSize == 0 || buffer == NULL) {
+        // Not available, use defaults
+        ctx->buffer = NULL;
+        ctx->bufferLen = 0;
+        return parser_init_context_empty;
+    }
+
+    ctx->buffer = buffer;
+    ctx->bufferLen = bufferSize;
+
+    return parser_ok;
+}
+
+parser_error_t parser_init(parser_context_t *ctx, const uint8_t *buffer, size_t bufferSize) {
+    parser_error_t err = parser_init_context(ctx, buffer, bufferSize);
+    if (err != parser_ok)
+        return err;
+
+    return err;
+}
```

Comment:
A single `return err` is sufficient. No need for the `if` statement here :wink:

Reply:
Updated the code.